### PR TITLE
Use `donationMethod` instead of `paymentMethod` for crypto flow

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to https://github.com/opencollective/opencollective-frontend/pull/6851#pullrequestreview-767935694

This is with related to the above comment. Here I've renamed the `paymentMethod` to `donationMethod` to avoid any confusion with the other `paymentMethod` variables.

### CODE REVIEW

**Review by znarf — APPROVED** [review]
Not sure about donationMethod, but it's always much better than the confusing paymentMethod.

Good for me.

Maybe 'crypto' should be moved as a constant to make things even clearer, not a required change.

**Comment by SudharakaP** [comment]
> Not sure about donationMethod, but it's always much better than the confusing paymentMethod.

I couldn't think of anything better for the moment. If you have any suggestions better than `donationMethod` let me know. 

> Good for me.
> 
> Maybe 'crypto' should be moved as a constant to make things even clearer, not a required change.

Yes I've added `DONATION_METHOD.CRYPTO` as a constant and used that instead. 👍🏽 

https://github.com/opencollective/opencollective-frontend/blob/8db5e3fafdc856f6071a4f810d89cdacb03cb422/components/contribution-flow/constants.js#L16-L18

File: rewrites.js [hunk 1]
```diff
@@ -219,7 +219,7 @@ exports.REWRITES = [
   },
   // New Routes -> New flow
   {
-    source: `/:collectiveSlug/:verb(donate)/:paymentMethod(crypto)?/:step(${contributionFlowSteps})?`,
+    source: `/:collectiveSlug/:verb(donate)/:donationMethod(crypto)?/:step(${contributionFlowSteps})?`,
```
Comment (Betree) [inline rewrites.js:222]: Not a fan of `donationMethod` either. It's already an improvement so not a blocker on my side, but I can suggest this alternative name:


```suggestion
    source: `/:collectiveSlug/:verb(donate)/:paymentFlow(crypto)?/:step(${contributionFlowSteps})?`,
```

Comment (SudharakaP) [reply rewrites.js]: Thanks for the suggestion. I've changed that. 👍🏽

**Review by Betree — APPROVED** [review]
Looking good!

**Pull request merged**